### DISCUSSION TITLE
Nested providers and access hierarchy

### DISCUSSION THREAD

**Discussion by mo-lon** [pmndrs/jotai#2704] [Q&A]
I was wondering if there is a recommended approach for accessing stores/atoms that are not in the nearest provider. I'm aware of the general behaviour where components can only access atoms from the nearest provider in the hierarchy.

For example [codesandbox](https://codesandbox.io/p/sandbox/smoosh-bash-qjwg6w), the Child component can access tabAtom but not tabsAtom. 

Is there an elegant way without having to pass grandparent store down as prop or using custom context?

**Comment by dai-shi** [comment]
Check out `ScopeProvider` in [jotai-scope](https://github.com/jotaijs/jotai-scope) which @dmaskasky is primarily maintaining.
